Makefile

```diff
@@ -42,25 +43,26 @@ generate::
 	go generate ./pkg/codegen/docs/
 
 build::
-	go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${PROJECT}
+	cd pkg && go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${PROJECT}
 
 install::
-	GOBIN=$(PULUMI_BIN) go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${PROJECT}
+	cd pkg && GOBIN=$(PULUMI_BIN) go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${PROJECT}
 
 dist::
-	go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${PROJECT}
+	cd pkg && go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${PROJECT}
 
 lint::
 	for DIR in "examples" "pkg" "sdk" "tests" ; do \
 		pushd $$DIR && golangci-lint run -c ../.golangci.yml --deadline 5m && popd ; \
 	done
 
 test_fast::
-	$(GO_TEST_FAST) ${PROJECT_PKGS}
+	cd pkg && $(GO_TEST_FAST) ${PROJECT_PKGS}
 
 test_all::
-	$(GO_TEST) ${PROJECT_PKGS}
-	$(GO_TEST) -v -p=1 ${EXTRA_TEST_PKGS}
+	cd pkg && $(GO_TEST) ${PROJECT_PKGS}
+	cd examples && $(GO_TEST) -v -p=1 ${EXAMPLES_PKGS}
+	cd tests && $(GO_TEST) -v -p=1 ${TESTS_PKGS}
```

Comment:
This will slow down the build by not running these two sets of tests in parallel.  Is it the case that we have no other option since these no need to be `go test` runs in different folders with different `go.mod`s?

We may want to move `examples` into `tests` as follow up (the examples are really tests not examples anyway, and are misleading as a top level `examples` as they are not helpful for users).

Reply:
Sounds good. I'll move examples as a part of this change.

Reply:
I'm actually going to do this as a part of another change given that we're having some issues with `make build` that this changeset will resolve.